cmd/ndm_daemonset/probe/udevprobe.go

```diff
@@ -142,28 +144,49 @@ func (up *udevProbe) scan() error {
 			continue
 		}
 		if newUdevice.IsDisk() || newUdevice.IsParitition() {
-			uuid := newUdevice.GetUid()
-			disksUid = append(disksUid, uuid)
 			deviceDetails := &blockdevice.BlockDevice{}
-			deviceDetails.UUID = uuid
+			if up.controller.FeatureGates.IsEnabled(features.GPTBasedUUID) {
+				// all fields that is used to generate UUID to be filled whenever the event is
+				// generated. This is required so that even if none of the probes work, the
+				// system should be able to generate the UUID of the blockdevice
+				deviceDetails.DeviceAttributes.DeviceType = newUdevice.GetPropertyValue(libudevwrapper.UDEV_DEVTYPE)
+				deviceDetails.DeviceAttributes.WWN = newUdevice.GetPropertyValue(libudevwrapper.UDEV_WWN)
+				deviceDetails.PartitionInfo.PartitionTableUUID = newUdevice.GetPropertyValue(libudevwrapper.UDEV_PARTITION_TABLE_UUID)
+				deviceDetails.PartitionInfo.PartitionEntryUUID = newUdevice.GetPropertyValue(libudevwrapper.UDEV_PARTITION_UUID)
+				deviceDetails.FSInfo.FileSystemUUID = newUdevice.GetPropertyValue(libudevwrapper.UDEV_FS_UUID)
+			} else {
+				uuid := newUdevice.GetUid()
+				disksUid = append(disksUid, uuid)
+				deviceDetails.UUID = uuid
+			}
 			deviceDetails.SysPath = newUdevice.GetSyspath()
 			deviceDetails.DevPath = newUdevice.GetPath()
 			diskInfo = append(diskInfo, deviceDetails)
 
-			// get the dependents of the block device and log it
+			// get the dependents of the block device
 			devicePath := hierarchy.Device{
 				Path: deviceDetails.DevPath,
 			}
 			dependents, err := devicePath.GetDependents()
 			if err != nil {
 				klog.Error("error getting dependent devices for ", deviceDetails.DevPath)
 			} else {
+				deviceDetails.Partitions = dependents.Partitions
+				deviceDetails.Holders = dependents.Holders
+				deviceDetails.Parent = dependents.Parent
+				deviceDetails.Slaves = dependents.Slaves
 				klog.Infof("Dependents of %s : %+v", deviceDetails.DevPath, dependents)
 			}
 		}
 		newUdevice.UdevDeviceUnref()
 	}
-	up.controller.DeactivateStaleBlockDeviceResource(disksUid)
+	if !up.controller.FeatureGates.IsEnabled(features.GPTBasedUUID) {
+		up.controller.DeactivateStaleBlockDeviceResource(disksUid)
+	} else {
+		// TODO deactivate all the stale blockdevices
```

Comment:
we should handle it otherwise removed devices will remain active.

Reply:
yes, we should deactivate them. But when to deactivate? 

Earlier we used to deactivate only those which were not identified. 

Shall we deactivate  all resources on the given host first and then let the event handler update them as active one by one. Are there any corner cases for this approach.?

Reply:
@pawanpraka1  WDYT about this approach?

Reply:
hmm, deactivating all of them seems to be ok as if it was graceful shutdown they will be Inactive anyway. I don't see any corner case just that it will be extra etcd call for the devices that are currently attached one for making them Incative and other for making them active. What is the problem in calling DeactivateStaleBlockDeviceResource here? We can have disksUid map created for feature GPTBasedUUID) and pass it here?